src/process_utility.c

```diff
@@ -282,9 +283,20 @@ check_altertable_add_column_for_compressed(Hypertable *ht, ColumnDef *col)
 					has_notnull = true;
 					continue;
 				case CONSTR_DEFAULT:
-					/* since default expressions might trigger a table rewrite we
-					 * only allow Const here for now */
-					if (!IsA(constraint->raw_expr, A_Const))
+					/*
+					 * Since default expressions might trigger a table rewrite we
+					 * only allow Const here for now.
+					 */
+					if (IsA(constraint->raw_expr, TypeCast))
+					{
+					   /*
+						* Ignore error for boolean column as values like
+						* 'True' or 'False' are treated as TypeCast.
+						*/
+						char *name = strVal(llast(((TypeCast *) constraint->raw_expr)->typeName->names));
+						is_bool = strcasestr(name, "bool") ? true : false;
+					}
+					else if (!is_bool && (!IsA(constraint->raw_expr, A_Const)))
```

Comment:
Isn't `is_bool` always `false` since it's an `else if` so `is_bool` is never set?

Reply:
yes looks like the else shouldnt be there